### PR TITLE
Add support for printing Optional values in configuration

### PR DESCRIPTION
**Pull request by Praveen2112**

### CODE REVIEW

File: configuration/src/test/java/io/airlift/configuration/TestConfigurationInspector.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        List<String> defaultString = getOnlyElement(inspector.inspect(configurationFactory)).getAttributes()
+                .stream().map(ConfigurationInspector.ConfigAttribute::getDefaultValue).collect(toImmutableList());
+        assertEquals(defaultString, ImmutableList.of("null"));
+
+        List<String> currentString = getOnlyElement(inspector.inspect(configurationFactory)).getAttributes()
+                .stream().map(ConfigurationInspector.ConfigAttribute::getCurrentValue).collect(toImmutableList());
+        assertEquals(currentString, ImmutableList.of("string"));
+    }
+
+    public static class Config2
```
Comment (kokosing) [inline configuration/src/test/java/io/airlift/configuration/TestConfigurationInspector.java]: ConfigWithOptionalValue

File: configuration/src/test/java/io/airlift/configuration/TestConfigurationInspector.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    @Test
+    public void testConfigAttributesForOptional()
+    {
+        Map<String, String> properties = ImmutableMap.of("stringValue", "string");
+        ConfigurationFactory configurationFactory = new ConfigurationFactory(properties);
+        configurationFactory.registerConfigurationClasses(binder -> configBinder(binder).bindConfig(Config2.class));
+        ConfigurationInspector inspector = new ConfigurationInspector();
+
+        List<String> defaultString = getOnlyElement(inspector.inspect(configurationFactory)).getAttributes()
+                .stream().map(ConfigurationInspector.ConfigAttribute::getDefaultValue).collect(toImmutableList());
```
Comment (kokosing) [inline configuration/src/test/java/io/airlift/configuration/TestConfigurationInspector.java]:
```
.stream()
.map(...)
.collect(...);
```

File: configuration/src/test/java/io/airlift/configuration/TestConfigurationInspector.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        assertEquals(defaultString, ImmutableList.of("null"));
+
+        List<String> currentString = getOnlyElement(inspector.inspect(configurationFactory)).getAttributes()
+                .stream().map(ConfigurationInspector.ConfigAttribute::getCurrentValue).collect(toImmutableList());
```
Comment (kokosing) [inline configuration/src/test/java/io/airlift/configuration/TestConfigurationInspector.java]: same here

File: configuration/src/test/java/io/airlift/configuration/TestConfigurationInspector.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        Map<String, String> properties = ImmutableMap.of("stringValue", "string");
+        ConfigurationFactory configurationFactory = new ConfigurationFactory(properties);
+        configurationFactory.registerConfigurationClasses(binder -> configBinder(binder).bindConfig(Config2.class));
+        ConfigurationInspector inspector = new ConfigurationInspector();
+
+        List<String> defaultString = getOnlyElement(inspector.inspect(configurationFactory)).getAttributes()
+                .stream().map(ConfigurationInspector.ConfigAttribute::getDefaultValue).collect(toImmutableList());
+        assertEquals(defaultString, ImmutableList.of("null"));
+
+        List<String> currentString = getOnlyElement(inspector.inspect(configurationFactory)).getAttributes()
```
Comment (kokosing) [inline configuration/src/test/java/io/airlift/configuration/TestConfigurationInspector.java]: nit: maybe lets split that into two tests and extract common code as methods?

**Comment by Praveen2112** [comment]
@kokosing Thanks for your review. Have applied your comments

**Comment by electrum** [comment]
@Praveen2112 I think you forgot to push as I don't see any changes

**Comment by Praveen2112** [comment]
@electrum Sorry I didn't pushed it. Now I have updated it

File: configuration/src/main/java/io/airlift/configuration/ConfigurationInspector.java [hunk 5]
```diff
@@ -52,6 +53,9 @@ private static String getValue(Method getter, Object instance, String defaultVal
             if (value == null) {
                 return "null";
             }
+            if (value instanceof Optional) {
+                return ((Optional<Object>) value).map(Object::toString).orElse("null");
```
Comment (findepi) [inline configuration/src/main/java/io/airlift/configuration/ConfigurationInspector.java]: This is an incorrect cast. Cast to `Optional<?>` instead.

BTW isn't printing `null` a bit inappropriate for non-null optional values?
Wouldn't `----` (or something) be better?
(We could change `"null"` above for `value == null` case too)

**Comment by Praveen2112** [comment]
@findepi Thanks for your insight have applied your comments

**Comment by electrum** [comment]
@Praveen2112 I fixed the Travis build, and now there's a legitimate failure:
```
[ERROR]   TestConfigurationInspector.testDefaultValue:27 Lists differ at element [0]: null != ---- expected [null] but found [----]
```

**Comment by Praveen2112** [comment]
@electrum Thanks for pointing it out. Have fixed that issue

**Comment by electrum** [comment]
Thanks!

**Pull request merged**